### DISCUSSION TITLE
workspaceState API in Extension Development Host

### DISCUSSION THREAD

**Discussion by mprotasov** [microsoft/vscode-discussions#515] [Extension Development QnA] [answered]
Hi! 

I'm using `ExtensionContext#workspaceState` API in my extension. 

Prior to VS Code 1.76.0, workspace in Extension Development Host on each re-run of extension during development (`F5`) was considered as a new workspace, so workspace storage was empty and reading value by key from `workspaceState `returned `undefined`

After update, this behaviour has been changed - when I try to read property by key from `workspaceState`, I receive value that was saved to storage during previous run. 

(When I close Extension Development Host, I always decline saving workspace configuration as a file)

Is it a bug or expected behaviour?

**Comment by digitarald** [comment]
If it isn't mentioned in the release notes, it maybe wasn't intended; I am double-checking though. If its breaking your expectations, it is worth filing an issue.

**Reply by mprotasov** [reply]
@digitarald, thanks for your response!

I double-checked logic on side of my extension, and now I'm not sure that it is an actual issue.

We can save data to `workspaceState` even if no workspace has been opened (`workspace.name === undefined`), and on next debug run we can read that data for not-opened-workspace.

I've added additional check in my code (`if (workspace.name) {...}`) for now. But maybe it is good idea to make docs more clear about this case since running extension host without workspace is default behaviour for debugging extensions in VS Code

**Comment by isidorn** [comment, marked as answer]
@mprotasov Are you still seeing this issue with the latest VS Code version?

**Reply by mprotasov** [reply]
@isidorn In latest VS Code version it works as expected now. Thanks!

**Reply by isidorn** [reply]
Thanks for letting us know. Thus closing this.
